src/hooks/useContextMenu.tsx

```diff
@@ -0,0 +1,35 @@
+import { useState, useCallback, useEffect } from 'react'
+
+export const useContextMenu = () => {
+  const [position, setPosition] = useState({ x: 0, y: 0, left: true })
+  const [isActive, setMenuActive] = useState(false)
+
+  useEffect(() => {
+    const closeContextMenu = () => setMenuActive(false)
+    document.addEventListener('click', closeContextMenu, true)
```

Comment:
I actually thought about this under shower today, maybe we could postpone adding the click handler until needed? Imagine this scenario: we display grid of 40 video previews, each of those has its own `ContextMenu` so on mount we create 40 click handlers, one for each. Shouldn't be too big of a problem I guess, but not ideal.

What if instead we added the click handler only once the menu gets opened? You can actually pass `{once: true}` to `addEventListener` so we wouldn't need to clean it up later. How does that sound?

Reply:
Hmmm, would be better that way, sounds good - let's do that